inprocess/src/main/java/io/grpc/inprocess/InProcessChannelBuilder.java

```diff
@@ -225,9 +222,23 @@ public InProcessChannelBuilder propagateCauseWithStatus(boolean enable) {
     return this;
   }
 
+  /**
+   * Sets whether to include the provided messageSize or not and is propagated
```

Comment:
The language here is pretty confusing. InProcessTransport is an implementation detail; it shouldn't be mentioned. How about:

> Assumes RPC messages are the specified size. This avoids serializing messages for metrics and retry memory tracking. This can dramatically improve performance when accurate message sizes are not needed and if nothing else needs the serialized message.

We should add a note to set this on both the client-side and server-side?